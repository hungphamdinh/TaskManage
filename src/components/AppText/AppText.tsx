import React from "react";
import { Text, Platform } from "react-native";
import { Colors, Fonts } from "../../themes";
const androidOS = Platform.OS === "android";
export default (props: any) => {
  const {
    text,
    style,
    color = Colors.appTextBlack,
    font = Fonts.type.regular,
    size = Fonts.size.regular,
    bold = false,
    italic = false,
    center = false,
    right = false,
    scale = true,
  } = props as any;

  return (
    <Text
      {...props}
      style={[
        // eslint-disable-next-line react-native/no-inline-styles
        {
          fontSize: size,
          color: color,
          fontFamily: font,
          fontWeight: bold ? (androidOS ? "bold" : "500") : "400",
          fontStyle: italic ? "italic" : "normal",
          textAlign: center ? "center" : right ? "right" : "left",
        },
        style,
      ]}
      allowFontScaling={scale}
    >
      {text}
    </Text>
  );
};
