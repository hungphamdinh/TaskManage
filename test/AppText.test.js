import React from "react";
import renderer from "react-test-renderer";

import AppText from "../src/components/AppText";

test("renders correctly", () => {
  const tree = renderer.create(<AppText text={"Hellozzz3232z"} />).toJSON();
  // tree.props.text;
  // tree.props.allowFontScaling;
  expect(tree).toMatchSnapshot();
});
