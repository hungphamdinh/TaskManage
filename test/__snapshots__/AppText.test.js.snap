// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  allowFontScaling={true}
  style={
    Array [
      Object {
        "color": "#000",
        "fontSize": 13,
        "fontStyle": "normal",
        "fontWeight": "400",
        "textAlign": "left",
      },
      undefined,
    ]
  }
  text="Hellozzz3232z"
>
  Hellozzz3232z
</Text>
`;
